src/ScheduleFunctions.cpp

```diff
@@ -322,14 +346,9 @@ Stmt build_provide_loop_nest_helper(string func_name,
 }
 
 // Build a loop nest about a provide node using a schedule
-Stmt build_provide_loop_nest(const map<string, Function> &env,
-                             string func_name,
-                             string prefix,
-                             int start_fuse,
-                             const vector<string> &dims,
-                             const FuncSchedule &f_sched,
-                             const Definition &def,
-                             bool is_update) {
+Stmt build_provide_loop_nest(const map<string, Function> &env, const string &prefix, const string &func_name,
```

Comment:
IMHO the previous one-per-line declaration style is much more readable. Also, the argument order changed for no obvious reason?

Reply:
Changed to match other argument lists that use the same set of arguments.

Reply:
I'll collapse argument lists onto one line. That's not an issue.

Reply:
@steven-johnson - some long parameter lists were wrapped, rather than chopped as they are here. Would you prefer one long line for such declarations, or is wrapping okay?

Reply:
Actually, [many](https://github.com/halide/Halide/blob/master/src/ScheduleFunctions.cpp#L76) [long](https://github.com/halide/Halide/blob/master/src/ScheduleFunctions.cpp#L325) [argument](https://github.com/halide/Halide/blob/master/src/ScheduleFunctions.cpp#L1158) [lists](https://github.com/halide/Halide/blob/master/src/ScheduleFunctions.cpp#L1266) [were](https://github.com/halide/Halide/blob/master/src/ScheduleFunctions.cpp#L1344) [varyingly](https://github.com/halide/Halide/blob/master/src/ScheduleFunctions.cpp#L1705) [chopped](https://github.com/halide/Halide/blob/master/src/ScheduleFunctions.cpp#L2045) or [wrapped](https://github.com/halide/Halide/blob/master/src/ScheduleFunctions.cpp#L2175) in the original code. I can take this opportunity to give the file a uniform style if you'd like. Whatever you prefer / is the project-wide guideline.

Reply:
Our semi-official policy is essentially "lines shouldn't be any longer than they need to be" -- there's no hard-wrap limit. (That said, it's pretty rare that a line needs to be > 120.) The rule above all else is "make it look like the existing style"; if you think that's a bad idea, include a reason why the style variation is justified.

Reply:
+1 to Steven's proposal

Reply:
Okay!

Reply:
QQ: do we prefer, A:
```
Stmt build_provide_loop_nest_helper(const string &func_name,
                                    const string &prefix,
                                    int start_fuse,
                                    const vector<string> &dims,
                                    const vector<Expr> &site,
                                    const vector<Expr> &values,
                                    const vector<Expr> &predicates,
                                    const FuncSchedule &func_s,
                                    const StageSchedule &stage_s,
                                    bool is_update) {
```
or B:
```
Stmt build_provide_loop_nest_helper(
    const string &func_name,
    const string &prefix,
    int start_fuse,
    const vector<string> &dims,
    const vector<Expr> &site,
    const vector<Expr> &values,
    const vector<Expr> &predicates,
    const FuncSchedule &func_s,
    const StageSchedule &stage_s,
    bool is_update) {
```
Or something else?

Reply:
@abadams @steven-johnson

Reply:
Oops, don't know why your comment didn't show up, Andrew.